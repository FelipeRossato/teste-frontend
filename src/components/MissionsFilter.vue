<template>
  <h1 
    id="missions" 
    :class="mode"
  >
    MISSÕES
  </h1>
  <div 
    class="filter-wrapper"
    :class="mode"
  >
    <input
      type="radio"
      name="Filter"
      v-model="selectedStatus" 
      value="ALL"
      id="todas"
    > 
    <p>
      <label for="todas">Todas</label>
    </p> 
    <input
      type="radio"
      name="Filter"
      v-model="selectedStatus" 
      value="IN_PROGRESS"
      id="em-andamento"
    > 
    <p>
      <label for="em-andamento">Em Andamento</label>
    </p> 
    <input
      type="radio"
      name="Filter"
      v-model="selectedStatus" 
      value="COMPLETED"
      id="completa"
    > 
    <p>
      <label for="completa">Completa</label>
    </p> 
    <input
      type="radio"
      name="Filter"
      v-model="selectedStatus" 
      value="NOT_STARTED"
      id="nao-iniciada"
    > 
    <p>
      <label for="nao-iniciada">Não Iniciada</label>
    </p> 
    <input
      type="radio"
      name="Filter"
      v-model="selectedStatus" 
      value="BLOCKED"
      id="bloqueada"
    > 
    <p>
      <label for="bloqueada">Bloqueada</label>
    </p>    
  </div>
  <div class="list-wrapper">
    <MissionsList 
      :filtered-missions="filteredMissions"
      :mode="mode"
    />
  </div>
</template>

<script>
import MissionsList from './MissionsList';

export default {
  name: 'MissionsFilter',
  components: {
    MissionsList
  },

  data() {
    return {
      selectedStatus: "ALL",
    }
  },

  props: {
    mode: {
      type: String,
      default: 'light'
    },
    missions: {
      type: Array,
      default: () => []
    }
  },
  
  computed: {
    // Função para filtrar as 'missions' por 'status'
    filteredMissions() {
      const status = this.selectedStatus;
      if (status === "ALL") {
        return this.missions
      } else {
        return this.missions.filter(mission => {
          return mission.status === status;
        })
      }
    }
  }
}
</script>

<style scoped>

  h1#missions {
    color: var(--title-color);
    font-weight: bold;
    font-size: 60px;
    margin-top: 30px;
    margin-left: 20px;
    position: relative;
  }

  h1#missions.dark {
    color: var(--main-color);
  }

  h1#missions::before {
    position: absolute;
    top: 59px;
    content: '';
    width: 280px;
    height: 10px;
    background-color: var(--background-color);
  }

  h1#missions.dark::before {
    background-color: var(--DM-background-color);
  }

  .filter-wrapper {
    height: 60px;
    width: 680px;
    margin: 35px 0 35px 20px;
    padding: 20px 40px;
    border-radius: 20px;
    background-color: var(--background-color);
    box-shadow: var(--box-shadow);
    transition: background-color 0.3s ease-in-out;
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-family: inherit;
    font-weight: bold;
    font-size: 15px;
    color: var(--text-color);
  }

  .filter-wrapper.dark {
    background-color: var(--DM-background-color);
    box-shadow: var(--DM-box-shadow);
    color: var(--DM-text-color);
  }

  input[type="radio"] {
    margin-right: -25px;
    appearance: none;
    background-color: var(--white-color);
    width: 15px;
    height: 15px;
    border-radius: 50%;
    cursor: pointer;
    outline: none;
  }

  input[type="radio"]:checked {
    background-color: var(--black-color);
  }

  .list-wrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
  }

  @media screen and (max-width: 720px) {
    .filter-wrapper {
      width: 660px;
    }   
  }
  
  @media screen and (max-width: 696px) {
    .filter-wrapper {
      display: grid;
      justify-self: start;
      grid-template-columns: 1fr 4fr;
      width: 180px;
      height: 200px;
      padding: 20px 10px;
    }

    input[type="radio"] {
      margin-right: 0;
    }
  }

</style>