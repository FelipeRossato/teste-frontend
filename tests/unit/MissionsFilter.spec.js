import { mount } from '@vue/test-utils'
import MissionsFilter from '../../src/components/MissionsFilter'

describe('MissionsFilter', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(MissionsFilter)
  })
  
  test('Renderização inicial e snapshot', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  test('Verifica se o filtro está funcionando', async () => {
    expect(wrapper.vm.selectedStatus).toBe('ALL')

    await wrapper.find('#em-andamento').trigger('click')
    expect(wrapper.vm.selectedStatus).toBe('IN_PROGRESS')

    await wrapper.find('#completa').trigger('click')
    expect(wrapper.vm.selectedStatus).toBe('COMPLETED')

    await wrapper.find('#nao-iniciada').trigger('click')
    expect(wrapper.vm.selectedStatus).toBe('NOT_STARTED')

    await wrapper.find('#bloqueada').trigger('click')
    expect(wrapper.vm.selectedStatus).toBe('BLOCKED')

    await wrapper.find('#todas').trigger('click')
    expect(wrapper.vm.selectedStatus).toBe('ALL')
  })
})