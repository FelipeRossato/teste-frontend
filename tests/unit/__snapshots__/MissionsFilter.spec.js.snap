// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsFilter Renderização inicial e snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__props": Array [],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      
      <h1
        class="light"
        id="missions"
      >
         MISSÕES 
      </h1>
      <div
        class="filter-wrapper light"
      >
        <input
          id="todas"
          name="Filter"
          type="radio"
          value="ALL"
        />
        <p>
          <label
            for="todas"
          >
            Todas
          </label>
        </p>
        <input
          id="em-andamento"
          name="Filter"
          type="radio"
          value="IN_PROGRESS"
        />
        <p>
          <label
            for="em-andamento"
          >
            Em Andamento
          </label>
        </p>
        <input
          id="completa"
          name="Filter"
          type="radio"
          value="COMPLETED"
        />
        <p>
          <label
            for="completa"
          >
            Completa
          </label>
        </p>
        <input
          id="nao-iniciada"
          name="Filter"
          type="radio"
          value="NOT_STARTED"
        />
        <p>
          <label
            for="nao-iniciada"
          >
            Não Iniciada
          </label>
        </p>
        <input
          id="bloqueada"
          name="Filter"
          type="radio"
          value="BLOCKED"
        />
        <p>
          <label
            for="bloqueada"
          >
            Bloqueada
          </label>
        </p>
      </div>
      <div
        class="list-wrapper"
      >
        
        
      </div>
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.6",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "filteredMissions": Array [],
    "missions": Array [],
    "mode": "light",
    "selectedStatus": "ALL",
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
